### ISSUE TITLE
`os.date` 2nd paramter (`time`) should be optional

### ISSUE THREAD

**Issue by Rathoz** [CppCXY/emmylua-analyzer-rust#109]
```lua
os.date('!%F') -- expected 2 but found 1. missing parameter: time EmmyLua(missing-parameter)
```

**Comment by rena-ertam028** [comment]
AI tried to solve the issue:

The `os.date` function's `time` parameter should be optional according to the documentation. Updated the function definition to make the `time` parameter optional by adding a `?` to its type annotation. This fixes the false positive missing-parameter warning when calling `os.date` with just the format parameter.

You can review changes in this commit: https://github.com/rena-ertam028/CppCXY-emmylua-analyzer-rust-109/commit/f71f979b8703e5ed882b8f3c395b77ff54c0d2b6.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.
